src/main/java/sirius/web/security/Permissions.java

```diff
@@ -157,4 +172,59 @@ public static boolean wearsPermissionAnnotation(AnnotatedElement object) {
 
         return object.isAnnotationPresent(LoginRequired.class);
     }
+
+    /**
+     * Determines if the permission expression is contained in the given permissions.
+     * <p>
+     * Next to plain permission names, permissions can also negated using <tt>!permission</tt> and on top of that,
+     * whole logical expressions in DNF (disjuctive normal form) can be passed in.
+     * <p>
+     * Such a formula is a set of expressions where a <b>,</b> represents an <tt>or</tt> and a <b>+</b> represents an
+     * <tt>and</tt>. An example would be "logged-in,important-customer+!locked". This would translate to "the user has
+     * to be logged in or it has to be an important customer and not be locked".
+     *
+     * @param permissionExpression the permission expression to check
+     * @param permissions          all permissions as set
+     * @return <tt>true</tt> if the given permissions contains the permission expression, <tt>false</tt> otherwise
+     */
+    @SuppressWarnings("squid:S2259")
+    @Explain("permissionExpression can not be null due to Strings.isEmpty")
+    public static boolean hasPermission(@Nullable String permissionExpression, @Nullable Set<String> permissions) {
```

Comment:
I am planning to refactor the "hasRequiredPermissionForPermission" of (this)[https://github.com/scireum/sirius-biz/pull/350] PR to take a Set instead of a Predicate, so this method is usable there

Reply:
no need to implement an `hasPermission(@Nullable String permissionExpression, @Nullable Predicate<String> hasPermission)` in my opinion

Reply:
no need for an implementation imposing a Set, Set::contains is an off the shelf impl for said prediacte.